flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/windowing/WindowOperatorTest.java

```diff
@@ -98,7 +99,13 @@
 	// late arriving event OutputTag<StreamRecord<IN>>
 	private static final OutputTag<Tuple2<String, Integer>> lateOutputTag = new OutputTag<Tuple2<String, Integer>>("late-output") {};
 
-	private void testSlidingEventTimeWindows(OneInputStreamOperatorTestHarness<Tuple2<String, Integer>, Tuple2<String, Integer>> testHarness) throws Exception {
+	private void testSlidingEventTimeWindows(OneInputStreamOperator<Tuple2<String, Integer>, Tuple2<String, Integer>> operator) throws Exception {
```

Comment:
Just a clarification of all the refactoring in this class (please let me know if I got anything wrong):

The changes here had 2 major objectives -
1. Always re-instantiate the test harness on the snapshot restore run.
2. Always remember to close the test harness at the end of each test (and, therefore, closing the environment properly).

Correct?

Reply:
Yes. Previously restoring tests were working on `closed` harness, which was working only because test harness it self wasn't properly closing itself 🙄